akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcProjectionImpl.scala

```diff
@@ -418,6 +570,95 @@ private[projection] object R2dbcProjectionImpl {
       implicit system: ActorSystem[_]): FlowWithContext[Envelope, ProjectionContext, Done, ProjectionContext, _] = {
     import R2dbcOffsetStore.Validation._
     implicit val ec: ExecutionContext = system.executionContext
+
+    // This is similar to DynamoDBProjectionImpl.replayIfPossible but difficult to extract common parts
+    // since this is flow processing
+    def replayIfPossible(originalEnvelope: Envelope): Future[Boolean] = {
+      val logPrefix = offsetStore.logPrefix
+      originalEnvelope match {
+        case originalEventEnvelope: EventEnvelope[Any @unchecked] if originalEventEnvelope.sequenceNr > 1 =>
+          val underlyingProvider = sourceProvider match {
+            case adapted: JavaToScalaBySliceSourceProviderAdapter[_, _] => adapted.delegate
+            case provider                                               => provider
+          }
+          underlyingProvider match {
+            case provider: LoadEventsByPersistenceIdSourceProvider[Any @unchecked]
+                if offsetStore.settings.replayOnRejectedSequenceNumbers =>
+              val persistenceId = originalEventEnvelope.persistenceId
+              offsetStore.storedSeqNr(persistenceId).flatMap { storedSeqNr =>
+                val fromSeqNr = storedSeqNr + 1
+                val toSeqNr = originalEventEnvelope.sequenceNr
+                logReplayRejected(logPrefix, persistenceId, fromSeqNr, toSeqNr)
```

Comment:
note to self: we should include the origin source in the log message, eg. pubsub will happen rather frequently

Reply:
That would be useful.

Reply:
added in https://github.com/akka/akka-projection/pull/1289/commits/d6ab41486fe95069adf203bee834d544b998d036